### DISCUSSION TITLE
Built-in input validation rule for enum values

### DISCUSSION THREAD

**Discussion by SeinopSys** [laravel/framework#39897] [Ideas] [answered]
When validating user input you may want to ensure that a value in the request comes from a list of supported values. This list has been made easier to define with PHP 8.1 in the form of native enums, more specifically backed enums:

```php
enum Suit: string
{
    case Hearts = 'H';
    case Diamonds = 'D';
    case Clubs = 'C';
    case Spades = 'S';
}
```

If I want to validate that a property in the request body matches one of these values, I can currently do that with the `Rule::in` validation like so:

```php
Rule:in(array_column(Suit::cases(), 'value'))
```

This gets all the cases for the enum, then collects their value properties in an array, which is compatible with `Rule::in`'s existing behavior. I'd like to request that this be added as a less verbose validation method, for example:

```php
Rule:enumValue(Suit::class)
```

This would allow the property to be either `'H'`, `'D'`, `'C'` or `'S'`. It would work in a similar way for `int` type backed enums.

**Comment by dvigueras** [comment]
If this feature request is accepted I can send a PR with the changes needed.

**Comment by JurianArie** [comment, marked as answer]
https://github.com/laravel/framework/releases/tag/v8.69.0 added an Enum validation rule

**Reply by SeinopSys** [reply]
Thank you, looks like #39437 implemented this already, this is great news!

**Comment by marcoaureliocardoso** [comment]
Enum Validation Rules only works with the BackedEnum Interface
https://laracasts.com/index.php/index.php/discuss/channels/laravel/enum-validation-rules-only-works-by-backedenum-interface

**Reply by henzeb** [reply]
You can use my package to add from/tryFrom to your enums you'd like to validate.

https://packagist.org/packages/henzeb/enumhancer
